readthedocs/core/utils/contact.py

```diff
@@ -4,71 +4,53 @@
 from django.conf import settings
 from django.core.mail import send_mail
 from django.template import Context, Engine
-from messages_extends import constants as message_constants
 
-from readthedocs.notifications import SiteNotification
-from readthedocs.notifications.backends import SiteBackend
 
 log = structlog.get_logger(__name__)
 
 
+# TODO: re-implement sending notifications to users.
+# This needs more thinking because the notifications where "Generic",
+# and now we need to register a  ``Message`` with its ID.
```

Comment:
This is a small utility script to send email _and_ site notifications to users. We used to contact them when deprecating stuffs and Santos originally created it for specific custom domain issues, IIRC.

I adapted it to keep sending email notifications, but I removed the site notifications for now because we need a `message_id` to store in the `Notification` object and the original sent messages are lost now --since they were temporal.

To keep using this script, we need to adapt it to:

1. receive a `message_id=` argument
2. create a new `Notification` object with the `message_id` passed
3. receive `attached_to` argument to know where to attach the notification to (e.g. `User`, `Project`, etc)

Then, the user/caller of this script will need to:

1. create a `Message` with the notification they want to send and register it using `registry.add(messages)`
2. call the script with this `message_id` and `attached_to`

This is required because now we are not storing the message into the database, but only the `message_id`. This means the message has to be _declared in the code_ at the moment of rendering the site notification to the user.

I will create an issue for this 👍🏼

Reply:
I opened https://github.com/readthedocs/readthedocs.org/issues/10986 to track this work.